views/js/pciCreator/dev/adaptiveChoiceInteraction/runtime/adaptiveChoiceInteraction.js

```diff
@@ -0,0 +1,132 @@
+            this.config = config || {};
+            this.$container = $(dom);
+            
+            //add method on(), off() and trigger() to the current object
+            event.addEventMgr(this);
+            
+            if (!pci.widgetRenderer) {
+                pci.widgetRenderer = new Renderer({
+                    serial : pci.serial,
+                    $container : this.$container,
+                    interaction : pci     
+                });
+                pci.widgetRenderer.setState('runtime');
+                pci.widgetRenderer.render(pci.properties);
+            }
+
+            //tell the rendering engine that I am ready
+            qtiCustomInteractionContext.notifyReady(this);
+        },
+        /**
+         * Programmatically set the response following the json schema described in
+         * http://www.imsglobal.org/assessment/pciv1p0cf/imsPCIv1p0cf.html#_Toc353965343
+         * 
+         * @param {Object} interaction
+         * @param {Object} response
+         */
+        setResponse : function (response) {
+            console.log(response);
+            if (response.list && _.isArray(response.list.integer)) {
+                _.forEach(response.list.integer, function(val) {
+                    this.$container.find('.js-answer-input[value="' + val + '"]').prop('checked', true);
+                });
+            }
+        },
+        /**
+         * Get the response in the json format described in
+         * http://www.imsglobal.org/assessment/pciv1p0cf/imsPCIv1p0cf.html#_Toc353965343
+         * 
+         * @param {Object} interaction
+         * @returns {Object}
+         */
+        getResponse : function () {
+            var response = [];
+            this.$container.find('.js-answer-input:checked').each(function () {
+                response.push(parseInt($(this).val(), 10));
+            });
+            return { "list": { "integer": response } };
+        },
+        /**
+         * Remove the current response set in the interaction
+         * The state may not be restored at this point.
+         * 
+         * @param {Object} interaction
+         */
+        resetResponse : function () {
+            this.$container.find('.js-answer-input:checked').each(function () {
+                $(this).prop('checked', false);
+            });
+        },
+        /**
+         * Reverse operation performed by render()
+         * After this function is executed, only the inital naked markup remains 
+         * Event listeners are removed and the state and the response are reset
+         * 
+         * @param {Object} interaction
+         */
+        destroy : function () {
+            this.$container.off().empty();
+        },
+        /**
+         * Restore the state of the interaction from the serializedState.
+         * 
+         * @param {Object} interaction
+         * @param {Object} serializedState - json format
+         */
+        setSerializedState : function (state) {
```

Comment:
You should implement the state management (even though they're only wrapper to set/getResponse)